runtime/domain.c

```diff
@@ -1661,8 +1665,11 @@ static void domain_terminate (void)
   pthread_sigmask(SIG_BLOCK, &mask, NULL);
 #endif
 
-  // run the domain termination hook
+  /* Join ongoing systhreads, if necessary, and then run user-defined
```

Comment:
I'm not fond of the fact that you describe the call to a generic function (`caml_domain_stop_hook`) by talking about a specific instance (joining systhreads).